### PR TITLE
HTML & CSS Practice

### PR DESCRIPTION
**Pull request by martthaa**
### HTML & CSS Practice

[Demo](https://martthaa.github.io/sweetPopUp/) |
[Code base](https://github.com/martthaa/sweetPopUp)

The code is submitted in a dedicated feature branch.

Only code files are submitted.

### CODE REVIEW

**Comment by kasionio** [comment]
Hi Martha. I didn't find any files in PR, please check

**Comment by martthaa** [comment]
Hi @kasionio, in submissions folder? In PR I added a link [Code Base](https://github.com/martthaa/sweetPopUp) to the repository with the code.

**Comment by kasionio** [comment]
> Hi @kasionio, in submissions folder? In PR I added a link [Code Base](https://github.com/martthaa/sweetPopUp) to the repository with the code.

You should add files that need to be reviewed 
https://kottans.org/documentation/docs/doc/contribute

**Comment by kasionio** [comment]
Still nothing 🤷🏼‍♂️

**Comment by martthaa** [comment]
привіт, @kasionio ) перевірте, будь ласка, чи все гаразд. Я старалась розібратись чому в мене в репозиторії була сіра папка з білою стрілочкою...і коли я натискала на папку, вона не відкривалася, але на моїй локальній машині все було гаразд, вміст був, але в github я не могла отримати до нього доступ.  Як я пізніше дізналась це - папка (локально) є вкладеним Git-репозиторієм , дерево якого SHA1 записується як «gitlink» (сіра папка з прямою білою стрілкою). Я довго пробувала вирішити це, але не вийшло( тому спробувала додати фали через інтерфейс git...ось посилання на мою [папку](https://github.com/martthaa/frontend-2022-homeworks/tree/html_css_Practice/submissions/martthaa/popUp)

**Comment by kasionio** [comment]
Let's also practice English :) Stackoverflow is always ready to help you. First, read it carefully and decide how you can solve it. If you'll still have questions, ask here
https://stackoverflow.com/questions/62056294/github-folders-have-a-white-arrow-on-them

**Comment by martthaa** [comment]
ok) thanks for the advice

**Comment by martthaa** [comment]
but you will review these files? Or I should I try it again via git ?

**Comment by kasionio** [comment]
> but you will review these files? Or I should I try it again via git ?

You should fix it anyway, and I'll look at your PR anyway also :)

**Review by kasionio — CHANGES_REQUESTED** [review]
Great job, cool design, I really like it! 
Please take a look at the comments and screenshots. And try to use a keyboard, there is no focus on the popup button. Also, try to improve naming, `box`, `last`, `more` is not good because sounds too general, don't be afraid to use more than word :) You can read [this](https://www.freecodecamp.org/news/css-naming-conventions-that-will-save-you-hours-of-debugging-35cea737d849/) article, but don't take it too seriously, it is just one point of view.
image
image

**Comment by kasionio** [comment]
And try to resolve issue with git, it is mandatory

**Comment by martthaa** [comment]
I realized my mistake, when I added files, I simply copied the folder that I had previously added to to git repository. As a result, I seemed to want to create a repository in the middle of the repository, which is not correct. Today I already added a new project DOM and it's actually a piece of cake to add folder to an existing repository, I don't know why it was so difficult for me last time. Thanks for the corrections, tips and advice. I will try to fix everything tomorrow

**Comment by kasionio** [comment]
No need to hurry, take your time. Anyway, I'll check it on Monday I guess

**Comment by martthaa** [comment]
Done : )

**Comment by kasionio** [comment]
> 

Take a look also in mobile view, now it looks not usable :(
Screenshot_2022-08-26-22-13-37-641_com android chrome

**Comment by martthaa** [comment]
[DEMO](https://martthaa.github.io/sweetPopUp/)

**Comment by kasionio** [comment]
Once you are done with the changes, please re-request review
image

**Comment by martthaa** [comment]
Make any other changes? or just click re-request review?

**Comment by kasionio** [comment]
I can't reach the popup from the keyboard now, as I don't see an active state on it. Also seems that some style fixes needed 
image

**Comment by martthaa** [comment]
I hope everything will be ok, [Demo](https://martthaa.github.io/sweetPopUp/)

File: submissions/martthaa/popUp/index.html [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    <meta name="viewport" content="width=device-width, initial-scale=1" />
+    <title>Sweet PopUp</title>
+    <link rel="shortcut icon" href="./src/img/dessert_donut_icon.png">
+    <link rel="stylesheet" href="./src/style.css" type="text/css" />
+  </head>
+  <body>
+    <header>
+      <nav class="nav">
+        <ul class="boxOfNavigationButtons">
+          <li class="list-item icon-recipes" tabindex="1">
```
Comment (kasionio) [inline submissions/martthaa/popUp/index.html:15]: tabindex is not needed within this task

File: submissions/martthaa/popUp/style.css [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+.popup_link {
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  align-items: center;
+  gap: 10px;
+  font-family: 'Brush Script MT', cursive;
+}
+
+.innerText{
```
[the remark this answers is not in the document]
Comment (kasionio) [reply submissions/martthaa/popUp/style.css:137]: still not implemented

File: submissions/martthaa/popUp/style.css [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+    color: 808080;
+}
+
+.popup_more:checked ~ .popup_btn {
+  opacity: 0;
+  visibility: hidden;
+}
+
+a:hover {
+  color: 808080;
```
Comment (kasionio) [inline submissions/martthaa/popUp/style.css:164]: missed `#`?

File: submissions/martthaa/popUp/index.html [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+    <meta name="author" content="Marta Sobotyk"/>
+    <meta name="viewport" content="width=device-width, initial-scale=1" />
+    <title>Sweet PopUp</title>
+    <link rel="shortcut icon" href="./src/img/dessert_donut_icon.png">
+    <link rel="stylesheet" href="./src/style.css" type="text/css" />
+  </head>
+  <body>
+    <header>
+      <nav class="nav">
+        <ul class="boxOfNavigationButtons">
```
Comment (kasionio) [inline submissions/martthaa/popUp/index.html:14]: why camelCase?

**Review by kasionio — CHANGES_REQUESTED** [review]
please look at the padding 
image

**Comment by martthaa** [comment]
Hi) about span:
 I didn't delete it because I use it to give proper text size. When I tried to give a font-size in parent class it causeв to photo resizing also. I tried to use :not(.button-icon) for parent class but it didn't help so for now

**Comment by martthaa** [comment]
[Demo](https://martthaa.github.io/sweetPopUp/)

**Review by kasionio — CHANGES_REQUESTED** [review]
1. Containers on links jumping on hover
2. No pointer on hover on the popup button
3. Clicking the tab at the keyboard leads to the link at the bottom, which is not expected behavior 
4. `More` button has no focus state
5. Weird focus on link, looks below 

image

**Comment by martthaa** [comment]
[Demo](https://martthaa.github.io/sweetPopUp/)

**Comment by kasionio** [comment]
> Hi) about span: I didn't delete it because I use it to give proper text size. When I tried to give a font-size in parent class it causeв to photo resizing also. I tried to use :not(.button-icon) for parent class but it didn't help so for now

You need to understand what `em` unit means. Please try to read any article about it and you will be able to remove it :) Also, `rem` is widely used today, read about it

File: submissions/martthaa/popUp/style.css [hunk 5]
```diff
@@ -0,0 +1,232 @@
...
+
+.avatar {
+  width: 65px;
+  height: 65px;
+  border-radius: 50%;
+}
+
+@media screen and (max-width:540px) {
+  .popup{
+    right: -240%;
```
Comment (kasionio) [inline submissions/martthaa/popUp/style.css:189]: why percents here and px in other @media below at the same time?

File: submissions/martthaa/popUp/style.css [hunk 6]
```diff
@@ -0,0 +1,232 @@
...
+  }
+}
+
+@media screen and (max-width:345px) {
+  .popup{
+    left: -115px;
+  }
+}
+
+
```
Comment (kasionio) [inline submissions/martthaa/popUp/style.css:230]: too many empty lines :)

File: submissions/martthaa/popUp/style.css (hunk 5 above)
Comment (kasionio) [reply submissions/martthaa/popUp/style.css:189]: with your current design seems you don't need to change popup right and top

**Review by kasionio — CHANGES_REQUESTED** [review]

**Comment by martthaa** [comment]
[DEMO](https://martthaa.github.io/sweetPopUp/)

File: submissions/martthaa/popUp/style.css [hunk 7]
```diff
@@ -0,0 +1,224 @@
...
+    width: 300px;
+    max-height: 480px;
+  }
+
+  .last-item-menu{
+    margin-top: 35px;
+  }
+}
+
+@media screen and (max-width:375px) {
```
Comment (kasionio) [inline submissions/martthaa/popUp/style.css:211]: why do you need so many breakpoints? it is a good practice to use several common breakpoints
https://www.lambdatest.com/blog/how-to-use-css-breakpoints-for-responsive-design/

**Review by kasionio — CHANGES_REQUESTED** [review]
I don't see a focus state on `More` button when using a keyboard

**Comment by martthaa** [comment]
> I don't see a focus state on `More` button when using a keyboard
image

**Review by kasionio — CHANGES_REQUESTED** [review]
Please try to click Space to open an additional menu, nothing happens

**Comment by martthaa** [comment]
> Please try to click Space to open an additional menu, nothing happens
actually, I don't understand why it doesn't work (
image

**Comment by kasionio** [comment]
> > Please try to click Space to open an additional menu, nothing happens
> > actually, I don't understand why it doesn't work (
> > image

Please try to look through HTML and dev console and try to understand how your code works

**Pull request closed without merging**